### PR TITLE
fix!: disable rate limiting completely by default

### PR DESCRIPTION
**Pull request by gdsoumya**
Related https://github.com/argoproj/argo-cd/issues/17257

This PR disabled the bucket limiter by default, setting the QPS(r) value to MaxFloat64 according to documentation of the limiter should ignore the bucket size completely [refer here](https://cs.opensource.google/go/x/time/+/refs/tags/v0.5.0:rate/rate.go;l=35-38)

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by gdsoumya** [comment]
@crenshaw-dev / @csantanapr / @jessesuen

File: cmd/argocd-application-controller/commands/argocd_application_controller.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&shardingAlgorithm, "sharding-method", env.StringFromEnv(common.EnvControllerShardingAlgorithm, common.DefaultShardingAlgorithm), "Enables choice of sharding method. Supported sharding methods are : [legacy, round-robin] ")
 	// global queue rate limit config
 	command.Flags().Int64Var(&workqueueRateLimit.BucketSize, "wq-bucket-size", env.ParseInt64FromEnv("WORKQUEUE_BUCKET_SIZE", 500, 1, math.MaxInt64), "Set Workqueue Rate Limiter Bucket Size, default 500")
-	command.Flags().Int64Var(&workqueueRateLimit.BucketQPS, "wq-bucket-qps", env.ParseInt64FromEnv("WORKQUEUE_BUCKET_QPS", 50, 1, math.MaxInt64), "Set Workqueue Rate Limiter Bucket QPS, default 50")
+	command.Flags().Float64Var(&workqueueRateLimit.BucketQPS, "wq-bucket-qps", env.ParseFloat64FromEnv("WORKQUEUE_BUCKET_QPS", math.MaxFloat64, 1, math.MaxFloat64), "Set Workqueue Rate Limiter Bucket QPS, default set to MaxFloat64 which disables the bucket limiter")
```
Comment (csantanapr) [inline cmd/argocd-application-controller/commands/argocd_application_controller.go:223]: Setting the value here, this is the value that would disable rate limit according to the library being used

https://cs.opensource.google/go/x/time/+/refs/tags/v0.5.0:rate/rate.go;l=35-38

// Inf is the infinite rate limit; it allows all events (even if burst is zero).
const Inf = Limit(math.MaxFloat64)

**Review by csantanapr — APPROVED** [review]

**Comment by Enclavet** [comment]
Change looks good. I did some testing with 4k apps and syncing back to back were consistent at 13 mins. Also reconciliation was normal.

**Comment by jannfis** [comment]
This is a breaking change, as it changes default behavior. 

From my point of view, making this opt-in is the correct way to do, though.

**Review by jannfis — APPROVED** [review]
LGTM

**Comment by jannfis** [comment]
There were no objections, neither here or on Slack.

Merging and cherry-picking into 2.10

**Pull request merged**